### PR TITLE
Using the new ConversationItemsList

### PR DESCRIPTION
**Pull request by thecombatwombat**
Refer to https://github.com/layerhq/Atlas-Android/pull/238 for corresponding LayerUI changes

### CODE REVIEW

File: app/src/main/java/com/layer/messenger/ConversationsListActivity.java [hunk 1]
```diff
@@ -1,118 +1,174 @@
...
 import com.layer.ui.util.views.SwipeableItem;
 
-public class ConversationsListActivity extends BaseActivity {
-
-    private ConversationsRecyclerView mConversationsList;
-
-    public ConversationsListActivity() {
-        super(R.layout.activity_conversations_list, R.menu.menu_conversations_list, R.string.title_conversations_list, false);
-    }
+public class ConversationsListActivity extends AppCompatActivity {
```
Comment (smpete) [inline app/src/main/java/com/layer/messenger/ConversationsListActivity.java]: Are we getting rid of `BaseActivity`?

File: app/src/main/java/com/layer/messenger/ConversationsListActivity.java [hunk 2]
```diff
@@ -1,118 +1,174 @@
...
+
+        binding.executePendingBindings();
     }
 
     @Override
-    protected void onDestroy() {
-        super.onDestroy();
-        if (mConversationsList != null) {
-            mConversationsList.onDestroy();
+    public void setTitle(CharSequence title) {
```
Comment (smpete) [inline app/src/main/java/com/layer/messenger/ConversationsListActivity.java]: For activities, I'd prefer to see all lifecycle methods before other methods. It makes the lifecycle easier to understand

Comment (thecombatwombat) [inline app/src/main/java/com/layer/messenger/ConversationsListActivity.java]: Done

File: app/src/main/java/com/layer/messenger/ConversationsListActivity.java (hunk 1 above)
Comment (thecombatwombat) [inline app/src/main/java/com/layer/messenger/ConversationsListActivity.java]: Maybe. The `BaseActivity` handles some boilerplate around `findViewById` that we're getting rid of with data binding. 

The rest of the methods are simple enough that repeating them is not that big a problem. 

That being said, once we have all these things converted over to use data binding, if we find BaseActivity useful enough then, we won't get rid of it. But for now, the newer layer ui classes won't work with BaseActivity.

**Pull request merged**